src/mlte/measurement/process_measurement.py

```diff
@@ -0,0 +1,94 @@
+"""
+Base class for measurement of external processes asynchronally.
+"""
+
+from __future__ import annotations
+
+import threading
+import time
+from typing import List
+
+from .measurement import Measurement
+from .result import Result
+from mlte._private.platform import is_windows
+from mlte._private import job
+
+# -----------------------------------------------------------------------------
+# ProcessMeasurement
+# -----------------------------------------------------------------------------
+
+class ProcessMeasurement(Measurement):
+    """Base class to be extended to measure external processes."""
+
+    @staticmethod
+    def start_process(script: str, arguments: List[str]) -> int:
+        """
+        Initialize an external process running training or similar script.
+
+        :param script: The full path to a Python script with the training or equivalent process to run.
+        :type script: str
+
+        :param arguments: A list of string arguments for the process.
+        :type arguments: List[str[]
+
+        :return: the id of the process that was created.
+        :rtype: int
+        """        
+        return job.spawn_python_job(script, arguments)
+
+    def __init__(self, instance: ProcessMeasurement, identifier: str):
+        """
+        Initialize a new ProcessMeasurement measurement.
+
+        :param identifier: A unique identifier for the measurement
+        :type identifier: str
+        """
+        super().__init__(instance, identifier)        
+        if is_windows():
```

Comment:
Do you not intend to support `ProcessMeasurement` on Windows? The checks from `LocalProcessCPUUtilization` and `LocalProcessMemoryConsumption` that `raise` on Windows are there because the particular way that we measure these metrics is specific to *nix platforms, but spawning a Python subprocess should work fine on both *nix and Windows (thanks to the abstractions Python's stdlib provides).

Reply:
That makes sense, I just copied that there before I fully implemented it, and didn't really review it it made sense later.